rayon-core/src/thread_pool/mod.rs

```diff
@@ -12,7 +12,49 @@ use std::error::Error;
 use registry::{Registry, WorkerThread};
 
 mod test;
-
+/// ## ThreadPool
+///
+/// `ThreadPool` represents a user created [threadpool]. [`ThreadPool::new()`]
+/// takes a [`Configuration`] struct that you can use to specify the number and/or
+/// names of threads in the pool.
+///
+/// Once the `ThreadPool` is created, there are two ways you can run jobs on it.
+/// You can use [`ThreadPool::install()`] to execute a closure in one of the
+/// threadpool's threads, or you can use [`ThreadPool::spawn_async()`]/[`ThreadPool::spawn_future_async()`].
+///
+///
+/// ## Creating a ThreadPool
+///
+/// ```rust
+///    # use rayon_core as rayon;
+///    use rayon::{ThreadPool, Configuration};
+///
+///    let pool = ThreadPool::new(Configuration::new().num_threads(8)).unwrap();
+/// ```
+///
+/// [`install()`] executes a closure in one of the `ThreadPool`'s threads. In addition, 
+/// any other rayon operations called inside of `install()` (such as `rayon::join()`)
+/// will also execute in the context of the `ThreadPool`.
+///
+/// The other way to execute a closure in the `ThreadPool` is to invoke [`ThreadPool::spawn_async()`]
```

Comment:
I think these methods are currently "feature-gated", and https://github.com/nikomatsakis/rayon/pull/313 renames them in any case. You might want to mention the unstable feature, or move the comments onto the method itself for now. 

In general, #313 adds a number of other methods here -- e.g., `join`, `spawn`, that seem relevant though. We should think about the best way to present them. The general idea is that all of the top-level functions that rayon offers will be mirrored in the `ThreadPool` API, which will then execute them in a specific thread-pool. (In contrast, the top-level functions execute (implicitly) in the *current* thread-pool.)